OpenRA.Mods.Common/Traits/AmmoPool.cs

```diff
@@ -131,11 +135,22 @@ public void Tick(Actor self)
 
 		public IEnumerable<PipType> GetPips(Actor self)
 		{
+			if (Info.ShowAsTextPip)
+				return null;
+
 			var pips = Info.PipCount >= 0 ? Info.PipCount : Info.Ammo;
 
 			return Enumerable.Range(0, pips).Select(i =>
 				(CurrentAmmo * pips) / Info.Ammo > i ?
 				Info.PipType : Info.PipTypeEmpty);
 		}
+
+		public Tuple<PipType, string> GetPip(Actor self)
+		{
+			if (!Info.ShowAsTextPip)
+				return null;
+
+			return Tuple.Create(Info.PipType, GetAmmoCount() + "/" + Info.Ammo);
```

Comment:
GetPip gets called a lot (typically 60 times per second), so you should cache and update the text label only when it changes to reduce memory churn.